spec/observables/dom/fetch-spec.ts

```diff
@@ -0,0 +1,195 @@
+  reset(): void;
+  respondWith: Response;
+}
+
+describe('fromFetch', () => {
+  let _fetch: typeof fetch;
+  let _AbortController: AbortController;
+
+  beforeEach(() => {
+    mockFetch.reset();
+    if (root.fetch) {
+      _fetch = root.fetch;
+    }
+    root.fetch = mockFetch;
+
+    MockAbortController.reset();
+    if (root.AbortController) {
+      _AbortController = root.AbortController;
+    }
+    root.AbortController = MockAbortController;
+  });
+
+  afterEach(() => {
+    root.fetch = _fetch;
+    root.AbortController = _AbortController;
+  });
+
+  it('should exist', () => {
+    expect(fromFetch).to.be.a('function');
+  });
+
+  it('should fetch', done => {
+    const fetch$ = fromFetch('/foo');
+    expect(mockFetch.calls.length).to.equal(0);
+    expect(MockAbortController.created).to.equal(0);
+
+    fetch$.subscribe({
+      next: response => {
+        expect(response).to.equal(OK_RESPONSE);
+      },
+      error: done,
+      complete: done,
+    });
+
+    expect(MockAbortController.created).to.equal(1);
+    expect(mockFetch.calls.length).to.equal(1);
+    expect(mockFetch.calls[0].input).to.equal('/foo');
+    expect(mockFetch.calls[0].init.signal).not.to.be.undefined;
+    expect(mockFetch.calls[0].init.signal.aborted).to.be.false;
+  });
+
+  it('should handle Response that is not `ok`', done => {
+    mockFetch.respondWith = {
+      ok: false,
+      status: 400,
+      body: 'Bad stuff here'
+    } as any as Response;
+
+    const fetch$ = fromFetch('/foo');
+    expect(mockFetch.calls.length).to.equal(0);
+    expect(MockAbortController.created).to.equal(0);
+
+    fetch$.subscribe({
+      next: () => done(new Error('should not be called')),
+      error: err => {
+        expect(err.message).to.equal('RxJS Fetch HTTP Error\n\nStatus:\n400\n\nBody:\nBad stuff here\n');
+        done();
+      },
+    });
+
+    expect(MockAbortController.created).to.equal(1);
+    expect(mockFetch.calls.length).to.equal(1);
+    expect(mockFetch.calls[0].input).to.equal('/foo');
+    expect(mockFetch.calls[0].init.signal).not.to.be.undefined;
+    expect(mockFetch.calls[0].init.signal.aborted).to.be.false;
+  });
+
+  it('should abort when unsubscribed', () => {
+    const fetch$ = fromFetch('/foo');
+    expect(mockFetch.calls.length).to.equal(0);
+    expect(MockAbortController.created).to.equal(0);
+    const subscription = fetch$.subscribe();
+
+    expect(MockAbortController.created).to.equal(1);
+    expect(mockFetch.calls.length).to.equal(1);
+    expect(mockFetch.calls[0].input).to.equal('/foo');
+    expect(mockFetch.calls[0].init.signal).not.to.be.undefined;
+    expect(mockFetch.calls[0].init.signal.aborted).to.be.false;
+
+    subscription.unsubscribe();
+    expect(mockFetch.calls[0].init.signal.aborted).to.be.true;
+  });
+
+  it('should allow passing of init object', () => {
+    const myInit = {};
+    const fetch$ = fromFetch('/foo', myInit);
+    fetch$.subscribe();
```

Comment:
Unlike the previous test - which calls `unsubscribe` - this one will have code that continues to execute after the test returns - as a `done` callback isn't used. That won't matter unless something gets broken in the implementation, but if it does, an error from this test will appear to come from someplace else.

I would pass a `done` callback here for `error` and `complete`.